packages/orchestrator/lib/routes/v1/postDequeue.ts

```diff
@@ -69,14 +69,13 @@ const handler = (scheduler: Scheduler, eventEmitter: EventEmitter) => {
             cleanupAndRespond((res) => res.status(200).send([]));
         }, longPollingTimeoutMs);
 
-        eventEmitter.once(eventId, onTaskStarted);
-
         const getTasks = await scheduler.dequeue({ groupKey, limit });
         if (getTasks.isErr()) {
             cleanupAndRespond((res) => res.status(500).json({ error: { code: 'dequeue_failed', message: getTasks.error.message } }));
             return;
         }
         if (longPolling && getTasks.value.length === 0) {
+            eventEmitter.once(eventId, onTaskStarted);
```

Comment:
Tasks could be dequeued at L72 but also in the event listener function on L61. But only one set of dequeued tasks will be returned, leaving the other tasks as started. They will eventually expires. 
The fix is simple: only listen to task creation event if long polling and initial dequeuing returned no tasks